lambdaclass/mirra_backend
# AddEnergy executions

## Motivation

Game design requires AddEnergy execution
Closes https://github.com/lambdaclass/champions_of_mirra/issues/400

## Summary of changes

- Implement AddEnergy execution
- Implement Energy capping at 500
- Implement tests for AddEnergy
- Fix preexisting tests

## How to test it?
Start a small fight with some unit with an AddEnergy skill in the console (check `battle.md` for examples, and `skills.json` to look for a healing character or to create your own). You should see a message getting logged when that unit casts its heal: "X adding N energy to Y (E -> E+N)". Remember that if your [info] logs are getting cut off, you can swap Logger.info calls with IO.inspect to debug
